influxdb3/src/commands/create.rs

```diff
@@ -0,0 +1,327 @@
+use crate::commands::common::{parse_key_val, DataType, InfluxDb3Config, SeparatedList};
+use base64::engine::general_purpose::URL_SAFE_NO_PAD as B64;
+use base64::Engine as _;
+use rand::rngs::OsRng;
+use rand::RngCore;
+use secrecy::ExposeSecret;
+use secrecy::Secret;
+use sha2::Digest;
+use sha2::Sha512;
+use std::error::Error;
+use std::num::NonZeroUsize;
+use std::str;
+use url::Url;
+
+#[derive(Debug, clap::Parser)]
+pub struct Config {
+    #[clap(subcommand)]
+    cmd: SubCommand,
+}
+
+#[derive(Debug, clap::Subcommand)]
+pub enum SubCommand {
+    /// Create a new database
+    Database(DatabaseConfig),
+    /// Create a new table in a database
+    Table(TableConfig),
+    /// Create a new auth token
+    Token,
+    /// Create a new last cache
+    LastCache(LastCacheConfig),
+    /// Create a new metacache
+    MetaCache(MetaCacheConfig),
+}
+
+#[derive(Debug, clap::Args)]
+pub struct DatabaseConfig {
+    /// The host URL of the running InfluxDB 3 Core server
+    #[clap(
+        short = 'H',
+        long = "host",
+        env = "INFLUXDB3_HOST_URL",
+        default_value = "http://127.0.0.1:8181"
+    )]
+    pub host_url: Url,
+
+    /// The token for authentication with the InfluxDB 3 Core server
+    #[clap(long = "token", env = "INFLUXDB3_AUTH_TOKEN")]
+    pub auth_token: Option<Secret<String>>,
+
+    /// The database name to run the query against
```

Comment:
This is a bit odd but if you run

```
influxdb3 create database -h
Create a new database

Usage: influxdb3 create database [OPTIONS] <DATABASE_NAME>

Arguments:
  <DATABASE_NAME>  The database name to run the query against [env: INFLUXDB3_DATABASE_NAME=]
```

The text is that you're running a query against a database, when if fact you're creating a database with that name. Ideally it would say that and also include some information about valid naming (i.e. alphanumeric with underscore and dash, starting with a letter or number).